### PR TITLE
[Merged by Bors] - feat: port Algebra.Quaternion

### PR DESCRIPTION
**Pull request by urkud**
---

The section about `DivisionRing` instance fails, even with η-experiment. I don't know how to fix it.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Quaternion.lean [hunk 1]
```diff
@@ -55,19 +55,19 @@ quaternion
 -/
 
 
-/- ./././Mathport/Syntax/Translate/Command.lean:429:34: infer kinds are unsupported in Lean 4: mk {} -/
 /-- Quaternion algebra over a type with fixed coefficients $a=i^2$ and $b=j^2$.
 Implemented as a structure with four fields: `re`, `im_i`, `im_j`, and `im_k`. -/
```
Comment (eric-wieser) [inline Mathlib/Algebra/Quaternion.lean:59]: Note these names in comments will need fixing.

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 2]
```diff
@@ -1716,6 +1716,10 @@ theorem toNat_cast (n : ℕ) : Cardinal.toNat n = n := by
   exact (Classical.choose_spec (lt_aleph0.1 (nat_lt_aleph0 n))).symm
 #align cardinal.to_nat_cast Cardinal.toNat_cast
 
+@[simp]
+theorem toNat_ofNat (n : ℕ) [n.AtLeastTwo] : Cardinal.toNat (OfNat.ofNat n) = OfNat.ofNat n :=
```
Comment (eric-wieser) [inline Mathlib/SetTheory/Cardinal/Basic.lean:1720]: Why `n.AtLeastTwo`?

Comment (urkud) [reply Mathlib/SetTheory/Cardinal/Basic.lean:1720]: See https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Nat/Cast/Defs.html#instOfNat

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**